src/cclib/parser/data.py

```diff
@@ -1,317 +1,328 @@
+                      means they are not specified in self._attrlist
+        """
+
+        if type(attributes) is not dict:
+            raise TypeError("attributes must be in a dictionary")
+
+        valid = [a for a in attributes if a in self._attrlist]
+        invalid = [a for a in attributes if a not in self._attrlist]
+
+        for attr in valid:
+            setattr(self, attr, attributes[attr])
+
+        self.arrayify()
+        self.typecheck()
+
+        return invalid
+
+    def typecheck(self):
+        """Check the types of all attributes.
+
+        If an attribute does not match the expected type, then attempt to
+        convert; if that fails, only then raise a TypeError.
+        """
+
+        self.arrayify()
+        for attr in [a for a in self._attrlist if hasattr(self, a)]:
+
+            val = getattr(self, attr)
+            if type(val) == self._attributes[attr].type:
+                continue
+
+            try:
+                val = self._attributes[attr].type(val)
+            except ValueError:
+                args = (attr, type(val), self._attributes[attr].type)
+                raise TypeError("attribute %s is %s instead of %s and could not be converted" % args)
+
+    def write(self, filename=None, *args, **kwargs):
+        """Write parsed attributes to a file.
+
+        Possible extensions:
+          .cjson or .json -  output a chemical JSON file
+          .cml - output a chemical markup language (CML) file
+          .xyz - output a Cartesian XYZ file of the last coordinates available
+        """
+
+        from ..writer import ccwrite
+        outputstr = ccwrite(self, outputdest=filename, *args, **kwargs)
+        return outputstr
+
+    def writejson(self, filename=None):
+        """Write parsed attributes to a JSON file."""
+        return self.write(filename=filename, outputtype='cjson')
+    def writecml(self, filename=None):
+        """Write parsed attributes to a CML file."""
+        return self.write(filename=filename, outputtype='cml')
+    def writexyz(self, filename=None):
+        """Write parsed attributes to an XML file."""
+        return self.write(filename=filename, outputtype='xyz')
+
+    @property
+    def stoichiometry(self):
+        return nuclear.Nuclear(self).stoichiometry()
+
+
+class ccData_optdone_bool(ccData):
+    """This is the version of ccData where optdone is a Boolean."""
+
+    def __init__(self, *args, **kwargs):
+
+        super(ccData_optdone_bool, self).__init__(*args, **kwargs)
+
+
+    def setattributes(self, *args, **kwargs):
+
+        invalid = super(ccData_optdone_bool, self).setattributes(*args, **kwargs)
+
+        # Reduce optdone to a Boolean, because it will be parsed as a list. If this list has any element,
+        # it means that there was an optimized structure and optdone should be True.
+        if hasattr(self, 'optdone'):
+            self.optdone = len(self.optdone) > 0
```

Comment:
Ok, I see it now. We should have a short page in the official documentation for a few simple guidelines, and one of them would be setting `git config --global core.autocrlf input` for Linux/OS X and `true` for Windows. I think that would avoid the problem of not all text editors obeying existing line endings.